src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,251 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+
+    private LocalDateTime parseDate(String date) {
+        LocalDateTime dateTime = LocalDateTime.parse(date, DateTimeFormatter.ofPattern("dd/M/yyyy Hmm"));
+        return dateTime;
+    }
+
+    /**
+     * Parses user input and returns the command.
+     *
+     * @param input User input.
+     * @return Command word.
+     */
+    public String parseCommand(String input) {
+        String[] command = input.split(" ");
+        return command[0];
+    }
+
+    /**
+     * Checks if the user input is formatted into a correct To-do task.
+     *
+     * @param input User input.
+     * @return Boolean.
+     */
+    public boolean canParseTodo(String input) {
+        String[] command = input.split(" ");
+        if (command.length == 1) {
+            return false;
+        } else if (command[1].equals("")) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    /**
+     * Parses the user input and returns the description of a To-do task.
+     *
+     * @param input User input.
+     * @return To-do description.
+     */
+    public String parseTodoDescription(String input) {
+        String[] command = input.split(" ");
+        String description = "";
+        for (int i = 1; i < command.length; i++) {
+            description += command[i] + " ";
+        }
+        return description;
+    }
+
+    /**
+     * Checks if the user input is formatted into a correct Deadline task.
+     *
+     * @param input User input.
+     * @return Boolean.
+     */
+    public boolean canParseDeadline(String input) {
+        String[] command = input.split(" ");
+        if (command.length == 1) {
+            return false;
+        } else if (command[1].equals("/by")) {
+            return false;
+        } else {
+            String[] details = input.split(" /by ");
+            if (details.length != 2) {
+                return false;
+            } else {
+                    return true;
+            }
+        }
+    }
+
+    /**
+     * Parses the user input and returns the description of a Deadline task.
+     *
+     * @param input User input.
+     * @return Deadline description.
+     */
+    public String parseDeadlineDescription(String input) {
+        String[] details = input.split(" /by ");
+        return details[0].substring(9);
```

Comment:
Might be good to avoid using magic numbers here. Instead of 9 perhaps you could have a constant string for `"deadline "` and run the length function on the string?